Assets/HoloToolkit/Input/Scripts/GazeManager.cs

```diff
@@ -100,13 +113,13 @@ private void UpdateRaycast()
             // Check if the currently hit object has changed
             if (oldFocusedObject != FocusedObject)
             {
-                if (oldFocusedObject != null)
+                if (oldFocusedObject != null && OnGazeExit != null)
                 {
-                    oldFocusedObject.SendMessage("OnGazeLeave", SendMessageOptions.DontRequireReceiver);
```

Comment:
IMO this kind of API change should only happen once we have proper release notes / versioning system in place. 
This is a major change and so far there's no good way to communicate this with people using the toolkit.
I think we should push this kind of changes very carefully.. those steps sound reasonable to me:
1) add the new feature but still default to the old one, update the release notes so people are aware of this new design pattern.
also let us know that in for example v 2.0 the code will be deprecated 
2) eventually iterate on 1.0 for a while adding non related changes
3) once it's time to deprecate the old system, add deprecate warnings like you did but keep the old stuff so people have time to convert their code
Obviously the release notes should be clear about this upcoming API change
4) once we move to 2.0 it's fine to remove the old code
Any thoughts?